sdk/identity/azure-identity/src/client_secret_credential.cpp

```diff
@@ -105,10 +105,7 @@ Azure::Core::Credentials::AccessToken ClientSecretCredential::GetToken(
           auto request = std::make_unique<TokenCredentialImpl::TokenRequest>(
               HttpMethod::Post, m_requestUrl, body.str());
 
-          if (m_isAdfs)
-          {
-            request->HttpRequest.SetHeader("Host", m_requestUrl.GetHost());
-          }
+          request->HttpRequest.SetHeader("Host", m_requestUrl.GetHost());
```

Comment:
Why do you need to manually set the Host field here? Isn't that automatically handled by the HTTP transport?

Reply:
One of them (must be WinHTTP) does not. But it is always safe to have it, that's why I removed the condition. ADFS was failing without it, `login.microsoft.com` was not. But if I add it, it does not fail either. So, no need to have it conditionally added.